src/sasa/sasa_HASEL.cpp

```diff
@@ -0,0 +1,2716 @@
+        }
+        if (AtomResidueName[1][i].compare("MET")==0) {
+          DeltaG[i].push_back (-3.807/1000*std::pow(T,2)+2.272*T-339.242);
+        }
+        if (AtomResidueName[1][i].compare("PHE")==0) {
+          DeltaG[i].push_back (-7.828/1000*std::pow(T,2)+4.644*T-687.134);
+        }
+        if (AtomResidueName[1][i].compare("PRO")==0) {
+          DeltaG[i].push_back (-2.347/1000*std::pow(T,2)+1.411*T-214.211);
+        }
+        if (AtomResidueName[1][i].compare("SER")==0) {
+          DeltaG[i].push_back (1.813/1000*std::pow(T,2)-1.05*T+150.289);
+        }
+        if (AtomResidueName[1][i].compare("THR")==0) {
+          DeltaG[i].push_back (-2.64/1000*std::pow(T,2)+1.591*T-240.267);
+        }
+        if (AtomResidueName[1][i].compare("TRP")==0) {
+          DeltaG[i].push_back (-11.66/1000*std::pow(T,2)+6.916*T-1024.284);
+        }
+        if (AtomResidueName[1][i].compare("TYR")==0) {
+          DeltaG[i].push_back (-7.513/1000*std::pow(T,2)+4.478*T-666.484);
+        }
+        if (AtomResidueName[1][i].compare("VAL")==0) {
+          DeltaG[i].push_back (-4.902/1000*std::pow(T,2)+2.921*T-433.013);
+        }
+        DeltaG[natoms].push_back (-0.6927/1000*std::pow(T,2)+0.4404*T-68.724);
+      }
+    }
+  }
+
+  Ti = T;
+
+  if (firstStepFlag ==0) {
+    if (approach!=2 && approach!=3) {
+      cout << "Unknown approach " << approach << endl;
+    }
+    for(unsigned i=0; i<natoms; i++) {
+      if (DeltaG[i].size()==0 ) {
+        cout << "Delta G value for residue " << AtomResidueName[1][i] << " not found " << endl;
+        error ("missing Delta G parameter\n");
+      }
+    }
+  }
+}
+
+
+//calculates neighbor list
+void SASA_HASEL::calcNlist() {
+  if(!nopbc) makeWhole();
+
+  for(unsigned i = 0; i < natoms; i++) {
+    Nlist[i].clear();
+  }
+
+  for(unsigned i = 0; i < natoms; i++) {
+    if (SASAparam[i].size()>0) {
+      for (unsigned j = 0; j < i; j++) {
+        if (SASAparam[j].size()>0) {
+          const Vector Delta_ij_vec = delta( getPosition(i), getPosition(j) );
+          double Delta_ij_mod = Delta_ij_vec.modulo()*10;
+          double overlapD = SASAparam[i][0]+SASAparam[j][0];
+          if (Delta_ij_mod < overlapD) {
+            Nlist.at(i).push_back (j);
+            Nlist.at(j).push_back (i);
+          }
+        }
+      }
+    }
+  }
+}
+
+
+//calculates SASA according to Hasel et al., Tetrahedron Computer Methodology Vol. 1, No. 2, pp. 103-116, 1988
+void SASA_HASEL::calculate() {
+  if(!nopbc) makeWhole();
+
+  if(getExchangeStep()) nl_update = 0;
+  if (firstStepFlag ==0) {
+    readPDB();
+    readSASAparam();
+  }
+  if (nl_update == 0) {
+    calcNlist();
+  }
+
+
+  auto* moldat = plumed.getActionSet().selectLatest<GenericMolInfo*>(this);
+  double Si, sasai, bij;
+  double sasa = 0;
+  vector<Vector> derivatives( natoms );
+  for(unsigned i = 0; i < natoms; i++) {
+    derivatives[i][0] = 0.;
+    derivatives[i][1] = 0.;
+    derivatives[i][2] = 0.;
+  }
+
```

Comment:
Also here I would recommend a check on moldat before using it

Reply:
I added that. Thanks for the suggestion